src/Controls/src/Core/Handlers/Items/Android/SizedItemContentView.cs

```diff
@@ -23,13 +24,31 @@ protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
 				return;
 			}
 
-			var targetWidth = _width();
-			var targetHeight = _height();
+			double targetWidth = _width();
+			double targetHeight = _height();
 
-			(Content.VirtualView as View).Measure(Context.FromPixels(targetWidth), Context.FromPixels(targetHeight),
-				MeasureFlags.IncludeMargins);
+			if (!double.IsInfinity(targetWidth))
+				targetWidth = (int)Context.FromPixels(targetWidth);
 
-			SetMeasuredDimension(targetWidth, targetHeight);
+			if (!double.IsInfinity(targetHeight))
+				targetHeight = (int)Context.FromPixels(targetHeight);
+
+			//_ = (Content.VirtualView as IView)
+			//	.Measure(targetWidth, targetHeight);
+
+			if (Content.VirtualView.Handler is IPlatformViewHandler pvh)
+			{
+				var widthSpec = Context.CreateMeasureSpec(targetWidth, 
+					double.IsInfinity(targetWidth) ? double.NaN : targetWidth
+					, targetWidth);
+				var heightSpec = Context.CreateMeasureSpec(targetHeight, double.IsInfinity(targetHeight) ? double.NaN : targetHeight
+					, targetHeight);
+
+				pvh.PlatformView.Measure(widthSpec, heightSpec);
```

Comment:
Things always seem to work better if you just call platform measure instead of trying to call `IView.Measure`.  If I change this to `IView.Measure` then things like `Frame` stop measuring correctly.

Though I don't think this mechanism of calling takes into account `Margins`. Not sure how to make all of that play together.

Reply:
@hartez  ?

Reply:
Anything that was still using MeasureFlags.IncludeMargins is suspect anyway. This is probably fine. If it's not, we'll fix it in post.